python/feathub/processors/flink/table_builder/over_window_utils.py

```diff
@@ -196,73 +135,71 @@ def _get_flink_over_window(
 
 
 def _get_over_window_agg_column_list(
+    table_schema: NativeFlinkTableSchema,
     window_descriptor: "OverWindowDescriptor",
     agg_descriptors: List["AggregationFieldDescriptor"],
 ) -> List[native_flink_expr.Expression]:
     return [
         _get_over_window_agg_select_expr(
-            native_flink_expr.call_sql(descriptor.expr),
+            descriptor,
             window_descriptor,
-            descriptor.agg_func,
+            table_schema.get_field_data_type(descriptor.field_name),
             "w",
         ).alias(descriptor.field_name)
         for descriptor in agg_descriptors
     ]
 
 
 def _get_over_window_agg_select_expr(
```

Comment:
It might be more readable to convert this method to a Java method, as it contains a lot of java method accesses.

Reply:
I added a util method in Java to get the `AggregateFunction` for the over window.